packages/jira-adapter/src/filters/issue_type_schemas/issue_type_scheme.ts

```diff
@@ -77,13 +77,27 @@ const deployIssueTypeIdsOrder = async (
 
 
 const deployIssueTypeSchema = async (
-  change: ModificationChange<InstanceElement>,
+  change: ModificationChange<InstanceElement> | AdditionChange<InstanceElement>,
   client: JiraClient,
   config: JiraConfig,
 ): Promise<void> => {
-  await defaultDeployChange({ change, client, apiDefinitions: config.apiDefinitions, fieldsToIgnore: ['issueTypeIds'] })
-  await deployNewAndDeletedIssueTypeIds(change, client)
-  await deployIssueTypeIdsOrder(change, client)
+  await defaultDeployChange({
+    change,
+    client,
+    apiDefinitions: config.apiDefinitions,
+    fieldsToIgnore: isModificationChange(change) ? ['issueTypeIds'] : [],
+  })
+
+  if (isAdditionChange(change)) {
+    change.data.after.value.id = change.data.after.value.issueTypeSchemeId
+    delete change.data.after.value.issueTypeSchemeId
+  }
+
+  if (isModificationChange(change)) {
+    const resolvedChange = await resolveChangeElement(change, getLookUpName)
+    await deployNewAndDeletedIssueTypeIds(resolvedChange, client)
+    await deployIssueTypeIdsOrder(resolvedChange, client)
+  }
```

Comment:
nit: I'm not sure the duplication-saving part here justifies the readability, maybe move these to the relevant `if`s without the ternary operator?
```suggestion
  if (isModificationChange(change)) {
    await defaultDeployChange({ change, client, apiDefinitions: config.apiDefinitions, fieldsToIgnore: ['issueTypeIds'] })
    const resolvedChange = await resolveChangeElement(change, getLookUpName)
    await deployNewAndDeletedIssueTypeIds(resolvedChange, client)
    await deployIssueTypeIdsOrder(resolvedChange, client)
    return
  }

  await defaultDeployChange({ change, client, apiDefinitions: config.apiDefinitions )}
  if (isAdditionChange(change)) {
    change.data.after.value.id = change.data.after.value.issueTypeSchemeId
    delete change.data.after.value.issueTypeSchemeId
  }

```